base/linalg/symmetric.jl

```diff
@@ -0,0 +1,64 @@
+# Symmetric matrices
+
+type Symmetric{T<:Number} <: AbstractMatrix{T}
+    S::Matrix{T}
+    uplo::Char
+end
+function Symmetric{T<:Number}(S::Matrix{T}, uplo::Symbol)
+    if size(S, 1) != size(S, 2) throw(DimensionMismatch("Matrix must be square")); end
+    return Symmetric(S, string(uplo)[1])
+end
+Symmetric(A::StridedMatrix) = Symmetric(A, :U)
+
+copy(A::Symmetric) = Symmetric(copy(A.S), A.uplo)
+size(A::Symmetric, args...) = size(A.S, args...)
+print_matrix(io::IO, A::Symmetric) = print_matrix(io, full(A))
+full(A::Symmetric) = A.S
+ishermitian{T<:Real}(A::Symmetric) = true
+ishermitian{T<:Complex}(A::Symmetric) = all(imag(A.S) .== 0)
+issym(A::Symmetric) = true
+transpose(A::Symmetric) = A
+
+*(A::Symmetric, B::Symmetric) = *(full(A), full(B))
+*(A::Symmetric, B::StridedMatrix) = *(full(A), B)
```

Comment:
I'm not completely sure but I think OpenBLAS' symmetric kernels used to be quite a bit slower than gemm and maybe that is the reason for this. We should revisit and probably use `symm` if it is not slower.